core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -232,6 +239,21 @@ public boolean isEnabled() {
       return enabled;
    }
 
+   private void checkStoreAvailability() {
+      if (!enabled) return;
+
+      if (!isAvailable()) {
```

Comment:
Couldn't this just check `unavailableException` if it is null or not? Seems we wouldn't need the `managerAvailable` variable as well then.